pkg/disttask/framework/storage/task_table.go

```diff
@@ -413,7 +413,8 @@ func (stm *TaskManager) FailTask(ctx context.Context, taskID int64, currentState
 		`update mysql.tidb_global_task
 			set state=%?,
 			    error = %?,
-			    state_update_time = CURRENT_TIMESTAMP()
+			    state_update_time = CURRENT_TIMESTAMP(),
+				end_time = CURRENT_TIMESTAMP()
```

Comment:
some terminate state of task is updated using `UpdateTaskAndAddSubTasks`, see where `.OnDone(` is called